### PR TITLE
Support custom scale for charts

### PR DESCRIPTION
**Pull request by DarkMatterMatt**
I wanted to be able to force my line chart scale to start at 0 so I did this 😄 

This is beautiful code to work with - it only took me ~30 mins to understand what I was looking at and edit it accordingly

### CODE REVIEW

**Comment by diogobernardino** [comment]
Oh, interesting to see how easily you hacked it.

Would you like to resolve conflicts and target `dev` branch instead? It's a potential feature to ship with v3.2.

**Comment by ubuntudroid** [comment]
@DarkMatterMatt any chance you could resolve the conflicts and get this ready to be merged? Looks like an interesting feature! :)

**Comment by DarkMatterMatt** [comment]
I think this is ready, what do you think @diogobernardino?

Sorry for the delay, this has been a bit of a low priority recently.

**Comment by diogobernardino** [comment]
Looks like we have more conflicts due to my recent push.

File: williamchart/src/main/java/com/db/williamchart/renderer/BarChartRenderer.kt [hunk 1]
```diff
@@ -178,7 +179,8 @@ class BarChartRenderer(
             dataPoint.screenPositionX = labelsLeftPosition + (widthBetweenLabels * index)
             dataPoint.screenPositionY =
                 innerFrame.bottom -
-                    (chartHeight * (dataPoint.value - scale.min) / scaleSize)
+                                   // bar length must be positive, or zero
+                    (chartHeight * max(0f, dataPoint.value - scale.min) / scaleSize)
```
Comment (DarkMatterMatt) [inline williamchart/src/main/java/com/db/williamchart/renderer/BarChartRenderer.kt]: If the bar length is negative then it goes off the chart and covers the labels

**Comment by DarkMatterMatt** [comment]
It's now available for all charts (line, bar & horizontal bar)

**Pull request merged**